### PR TITLE
Preserve original image extension in backend processing

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Jpegs were being converted to png always.

e2e test reflects this change.

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/image_utils.py [hunk 1]
```diff
@@ -27,7 +28,7 @@ def format_image(
         return np.array(im)
     elif type == "filepath":
         path = processing_utils.save_pil_to_cache(
-            im, cache_dir=cache_dir, name=name, format=fmt or "png"  # type: ignore
+            im, cache_dir=cache_dir, name=name, format=fmt or format  # type: ignore
```
Comment (abidlabs) [inline gradio/image_utils.py:31]: The only risk (admittedly small) I can see here is that there are some image formats that PIL can read but not save as: https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#read-only-formats

They all look very rare to me, but perhaps a more robust approach would be to try / catch this, and save as "png" as a fallback

File: js/app/test/image_component_events.spec.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ test("Image click-to-upload uploads image successfuly. Clear button dispatches e
 	await page.getByLabel("Download").click();
 	const download = await downloadPromise;
 	// Automatically convert to png in the backend since PIL is very picky
-	await expect(download.suggestedFilename()).toBe("cheetah1.png");
+	await expect(download.suggestedFilename()).toBe("cheetah1.jpeg");
```
Comment (abidlabs) [inline js/app/test/image_component_events.spec.ts:24]: nice

**Comment by abidlabs** [comment]
Awesome thanks for taking care of this @freddyaboulton. Left a small note, otherwise lgtm

File: gradio/image_utils.py (hunk 1 above)
Comment (freddyaboulton) [reply gradio/image_utils.py:31]: Yea that's a good point

**Comment by freddyaboulton** [comment]
Just pushed up suggested changes @pngwn and @abidlabs !

File: gradio/image_utils.py [hunk 3]
```diff
@@ -39,10 +46,21 @@ def format_image(
...
         path = processing_utils.save_img_array_to_cache(y, cache_dir=cache_dir)
     elif isinstance(y, _Image.Image):
-        path = processing_utils.save_pil_to_cache(y, cache_dir=cache_dir)
+        fmt = y.format
+        try:
+            path = processing_utils.save_pil_to_cache(
+                y, cache_dir=cache_dir, format=fmt  # type: ignore
+            )
+        # Catch error if format is not supported by PIL
+        except Exception:
```
Comment (abidlabs) [inline gradio/image_utils.py:60]: looks like its a KeyError that PIL raises for invalid formats
```suggestion
        except KeyError:
```

File: gradio/image_utils.py [hunk 4]
```diff
@@ -26,9 +27,15 @@ def format_image(
...
     elif type == "filepath":
-        path = processing_utils.save_pil_to_cache(
-            im, cache_dir=cache_dir, name=name, format=fmt or "png"  # type: ignore
-        )
+        try:
+            path = processing_utils.save_pil_to_cache(
+                im, cache_dir=cache_dir, name=name, format=fmt or format  # type: ignore
+            )
+        # Catch error if format is not supported by PIL
+        except Exception:
```
Comment (abidlabs) [inline gradio/image_utils.py:35]: looks like its a KeyError that PIL raises for invalid formats
```suggestion
        except KeyError:
```

**Comment by abidlabs** [comment]
Nice thanks @freddyaboulton!

**Review by abidlabs — APPROVED** [review]

File: gradio/image_utils.py (hunk 4 above)
Comment (freddyaboulton) [reply gradio/image_utils.py:35]: Is that the only one?

Comment (abidlabs) [reply gradio/image_utils.py:35]: For invalid formats, yes. Tested with:

```py
from PIL import Image

Image.open("cheetah.jpg").save("c2.jpg", format="cur")  # cur is a read only format
```

Comment (freddyaboulton) [reply gradio/image_utils.py:35]: Let's see what happens!

**Comment by freddyaboulton** [comment]
Thanks for the reviews @abidlabs @pngwn !

**Pull request merged**